compiler/rustc_lint/src/for_loops_over_fallibles.rs

```diff
@@ -139,9 +139,10 @@ fn suggest_question_mark<'tcx>(
 
     let ty = substs.type_at(0);
     let infcx = cx.tcx.infer_ctxt().build();
+    let body_def_id = cx.tcx.hir().body_owner_def_id(body_id);
```

Comment:
seems okayish, but we may want to look into changing `enclosing_body` in the future

Reply:
yes, I had this on my todo list for the followup PR this week!